### DISCUSSION TITLE
error installing libmesh on a HPC cluster (petsc-3.14.2)

### DISCUSSION THREAD

**Discussion by mcacace** [idaholab/moose#17040] [Q&A Getting Started] [answered]
Dear all,
We are having some problematic times trying to upgrade our moose/libmesh installation on our internal cluster - while installing libmesh. Attached the config.log file. 
Note: for the first time, I did not install manually petsc (3.14.2), but used the script from the framework. Had problems with the direct solver strumpack (deactivated), but I do not think it matters. 
Before re-installing manually everything, I was wondering if anyone could see what I am missing.
Thank in advance for any help,
mauro
![screen_shot_error](https://user-images.githubusercontent.com/13695024/108244719-c9fba780-714f-11eb-9bda-a27853cca054.png)
[config.log](https://github.com/idaholab/moose/files/5997548/config.log)

**Comment by mcacace** [comment]
Dear all,
I am re-opening this post, since I cannot find a solution to the problem above. 
I should add that I am able to compile an older version of the framework (specific commit 54c00d0) against petsc version 3.12.5 (manually compiled). However I cannot upgrade to the newest version. I am installing petsc (3.14.2) relying on the procedure as per the website. I checked the installation and all worked fine., but while trying to build libmesh it always fails with the same error as I posted above (I can link again the log file in case). 
It would be nice if anyone from the developer team and/or users group could help me sorting out this issue. 
Thanks in advance for any help,
Mauro

**Reply by mcacace** [reply]
Attached the whole build.log (>4 MB) file for libmesh in case it could be of any help (the config.log is as for the one above and it looks OK to me at least)
[build.log](https://github.com/idaholab/moose/files/6063764/build.log)

**Comment by GiudGiud** [comment]
Hi Mauro

Is this fixed? I ll tag people who may be able to help if not

**Comment by mcacace** [comment]
Thanks for your reply @GiudGiud. 
The problem still persists, though I did not (re)try to compile. This said, I would much appreciate any guidance in this respect. 
Looking forward to feedback.
Mauro

**Comment by milljm** [comment]
I saw this 28 days ago, and I still don't have an idea... @roystgnr or @lindsayad ?

**Reply by GiudGiud** [reply]
@fdkong was working on push_parallel_vectors lately he might have an idea too

**Comment by fdkong** [comment]
It could be an "old" compiler. Could you should the result of "mpicxx -show"?

**Comment by mcacace** [comment, marked as answer]
@fdkong I am loading gcc 7.2 on openmpi 4. Here is the show on the mpicxx:
g++ -I/cluster/openmpi/gcc-7.2.0/openmpi-4.0.1/include -pthread -Wl,-rpath -Wl,/cluster/openmpi/gcc-7.2.0/openmpi-4.0.1/lib -Wl,--enable-new-dtags -L/cluster/openmpi/gcc-7.2.0/openmpi-4.0.1/lib -lmpi

Should I try to see whether I can upgrade?

**Reply by roystgnr** [reply]
We should definitely still be supporting 7.2; if not then that's something that we should fix or workaround on our part.  But it couldn't hurt to try a newer compiler to confirm or rule that out.

**Reply by roystgnr** [reply]
Oh, wait!  Just looked at your build.log.  We've seen this once before - the trouble is likely that we've started requesting more recent C++ standards support by default, but some older compilers accept the flags to support newer standards, pass our configure-time tests with those flags on, but then fail to actually compile our code.

Try forcing an older standard by configuring libMesh with ```--cxx-std=2011``` and see if that works any better.

**Reply by lindsayad** [reply]
We really need @jwpeterson to work his m4 C++ standard testing magic

**Reply by roystgnr** [reply]
The trouble here is that I'm not even sure how we'd write a test for this problem.  It's not a missing feature that we're running into; it'seems to be a compiler bug that gets triggered by a very complex piece of code.  It'd be a ton of work to boil it down into a test small enough for an autoconf macro that still triggers the same problem.

I'd say we should just downgrade the level of C++ we ask for, since we don't need C++17 yet, but I've been hoping we could actually upgrade to need (and thus to be able to internally use) C++17 later this year, now that even the conservative RHEL distributions default to gcc 8.  But if people are still using gcc 7.2 and if it's responsible for this build failure then perhaps I'm being too optimistic?  gcc 7 is supposed to have full C++17 support too, in theory, but in practice here we are.

**Comment by mcacace** [comment]
@roystgnr and all other: Thanks! I really had some terrible time trying to figure out what I was doing wrong ... 
I will set up myself to it tomorrow (CET) and will report asap.

**Comment by mcacace** [comment]
After upgrading to gcc 9.2 libmesh install without any problem. @roystgnr: your guess was right. Though I thought I did try to force the 11 standards before, but surely I did mix things up. Thanks for the support.
